msgpack/unpack.scm

```diff
@@ -168,11 +171,16 @@
       (throw 'unexpected-eof))))
 
 
-(define (unpack-string size in)
-  "- Scheme Procedure: unpack-string size in
-     Unpack a UTF-8 string of length ‘size’ from binary input port ‘in’."
+(define (unpack-string-with-decoder size in proc)
+  "- Scheme Procedure: unpack-string-with-decoder size in proc
+     Unpack a string of length ‘size’ from binary input port ‘in’.
+
+    If proc is not #f, (‘proc’ bytevector) is called to decode the string,
+    otherwise it is decoded as utf8."
   (define bytes (get-bytevector-n in size))
-  (utf8->string bytes))
+  (if proc
+    (proc bytes)
+    (utf8->string bytes)))
```

Comment:
If you followed my advice above there will always be an appropriate `proc`

```suggestion
  (proc bytes))
```

As an aside, even if we wanted to keep the two possibilities we could have written the expression more concisely:

```suggestion
((or proc utf8->string) bytes))
```

This works because in Scheme everything everything except `#f` is truthy and `or` evaluates to its firth truthy argument.